react/features/base/participants/reducer.js

```diff
@@ -112,7 +117,11 @@ function _participant(state: Object = {}, action) {
             set(state, 'dominantSpeaker', state.id === action.participant.id));
 
     case PARTICIPANT_ID_CHANGED:
-        if (state.id === action.oldValue) {
+        // PARTICIPANT_ID_CHANGED is only utilized for the local participant at
+        // the time of this writing. Because there is a potential risk for
+        // participant ID collisions in Jitsi Meet in general, minimize the risk
+        // by ensuring that the modified state is for the local participant.
+        if (state.id === action.oldValue && state.local) {
```

Comment:
would it make sense to compare state's and action's 'local' flag value and set it to 'true' when local participant's id is to be changed instead of limiting it to local participant ? Otherwise I would suggest to rename the action to LOCAL_PARTICIPANT_ID_CHANGED, so that it would not be misleading.